### PR TITLE
Add functionality to change profile attributes via API

### PR DESCRIPTION
**Pull request by rvdpol**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,7 +22,7 @@
     "atob": "^2.1.2",
     "axios": "^0.19.2",
     "body-parser": "^1.19.0",
-    "dotenv": "^8.2.0",
+    "dotenv": "^6.0.0",
```
Comment (andymarch) [inline package.json]: Is there a reason you are dropping the version here?

File: index.js [hunk 2]
```diff
@@ -111,18 +114,144 @@ router.get("/", async (req, res, next) => {
 
 });
 
+router.get("/authorization-code/default",
```
Comment (andymarch) [inline index.js]: This should be handled by the tenantResolver registerTenantRoutes

File: tenantResolver.js [hunk 3]
```diff
@@ -4,12 +4,20 @@ var passport = require('passport');
 var OidcStrategy = require('passport-openidconnect').Strategy;
 var logger = require('./logger')
 
-const defaultTenantSub = "default"
+const defaultTenantSub = "default";
+
+var defaultTenantConfig = {
```
Comment (andymarch) [inline tenantResolver.js]: This is handled automatically by providing a null config, see Tenant constructor

File: index.js [hunk 4]
```diff
@@ -1,4 +1,6 @@
 require('dotenv').config()
+require('https').globalAgent.options.rejectUnauthorized = false;
```
Comment (andymarch) [inline index.js]: Is there a reason you are changing the TLS behaviour here, @frenkel34 using weird certs again?

File: tenantResolver.js (hunk 3 above)
Comment (andymarch) [reply tenantResolver.js]: The env you have for OKTA_ORG_URL should be TENANT, i think the rest should then just work. I'll commit an example env file. Sorry this should have been in the inital commit.

**Review by andymarch — CHANGES_REQUESTED** [review]
Couple of strange bits here, can you explain them. The changes you've made to handle the default tenant should not be necessary with the correct env file, I'll get this committed now

File: tenantResolver.js (hunk 3 above)
Comment (rvdpol) [reply tenantResolver.js]: Correct, thanks, removed the code

File: index.js (hunk 2 above)
Comment (rvdpol) [reply index.js]: Removed

File: index.js (hunk 4 above)
Comment (rvdpol) [reply index.js]: Yes its the only way I could get it to work. Removed it from repository, will use locally if needed.

**Pull request merged**